src/Adapter/Presenter/Cart/CartLazyArray.php

```diff
@@ -0,0 +1,646 @@
+        $subtotals['products'] = [
+            'type' => 'products',
+            'label' => $this->translator->trans('Subtotal', [], 'Shop.Theme.Checkout'),
+            'amount' => $totalCartAmount,
+            'value' => $this->priceFormatter->format($totalCartAmount),
+        ];
+        if ($total_discount) {
+            $subtotals['discounts'] = [
+                'type' => 'discount',
+                'label' => $this->translator->trans('Discount(s)', [], 'Shop.Theme.Checkout'),
+                'amount' => $total_discount,
+                'value' => $this->priceFormatter->format($total_discount),
+            ];
+        } else {
+            $subtotals['discounts'] = null;
+        }
+        if ($this->cart->gift) {
+            $giftWrappingPrice = ($this->cart->getGiftWrappingPrice($this->includeTaxes()) != 0)
+                ? $this->cart->getGiftWrappingPrice($this->includeTaxes())
+                : 0;
+
+            $subtotals['gift_wrapping'] = [
+                'type' => 'gift_wrapping',
+                'label' => $this->translator->trans('Gift wrapping', [], 'Shop.Theme.Checkout'),
+                'amount' => $giftWrappingPrice,
+                'value' => ($giftWrappingPrice > 0)
+                    ? $this->priceFormatter->convertAndFormat($giftWrappingPrice)
+                    : $this->translator->trans('Free', [], 'Shop.Theme.Checkout'),
+            ];
+        }
+        if (!$this->cart->isVirtualCart()) {
+            $shippingCost = $this->cart->getTotalShippingCost(null, $this->includeTaxes());
+        } else {
+            $shippingCost = 0;
+        }
+        $subtotals['shipping'] = [
+            'type' => 'shipping',
+            'label' => $this->translator->trans('Shipping', [], 'Shop.Theme.Checkout'),
+            'amount' => $shippingCost,
+            'value' => $this->getShippingDisplayValue($this->cart, $shippingCost),
+        ];
+        $subtotals['tax'] = null;
+        if (Configuration::get('PS_TAX_DISPLAY')) {
+            $total_excluding_tax = $this->cart->getOrderTotal(false);
+            $total_including_tax = $this->cart->getOrderTotal(true);
+            $taxAmount = $total_including_tax - $total_excluding_tax;
+            $subtotals['tax'] = [
+                'type' => 'tax',
+                'label' => ($this->includeTaxes())
+                    ? $this->translator->trans('Included taxes', [], 'Shop.Theme.Checkout')
+                    : $this->translator->trans('Taxes', [], 'Shop.Theme.Checkout'),
+                'amount' => $taxAmount,
+                'value' => $this->priceFormatter->format($taxAmount),
+            ];
+        }
+
+        $this->subTotals = $subtotals;
+
+        return $this->subTotals;
+    }
+
+    /**
+     * @arrayAccess
+     */
+    public function getProductsCount(): int
+    {
+        // If product list is already available, no need to execute a new sql query
+        if (!empty($this->products)) {
```

Comment:
You need to specificaly check for a null or !isset.

If the cart is empty, the condition will pass and execute the query again.

Reply:
oops, my bad 👍 fixed

(just in case, all getters are executed only once here, because the AbstractLazyArray caches getters's result, but yes indeed it would execute the count query when it's not needed 👍  )